core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/Fitter.java

```diff
@@ -0,0 +1,50 @@
+package ai.timefold.solver.core.impl.solver;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.api.solver.RecommendedFit;
+import ai.timefold.solver.core.api.solver.ScoreAnalysisFetchPolicy;
+import ai.timefold.solver.core.impl.score.director.InnerScoreDirector;
+
+final class Fitter<Solution_, In_, Out_, Score_ extends Score<Score_>>
+        implements Function<InnerScoreDirector<Solution_, Score_>, List<RecommendedFit<Out_, Score_>>> {
+
+    private final DefaultSolverFactory<Solution_> solverFactory;
+    private final Solution_ originalSolution;
+    private final In_ originalElement;
+    private final Function<In_, Out_> propositionFunction;
+    private final ScoreAnalysisFetchPolicy fetchPolicy;
+
+    public Fitter(DefaultSolverFactory<Solution_> solverFactory, Solution_ originalSolution, In_ originalElement,
+            Function<In_, Out_> propositionFunction, ScoreAnalysisFetchPolicy fetchPolicy) {
+        this.solverFactory = Objects.requireNonNull(solverFactory);
+        this.originalSolution = Objects.requireNonNull(originalSolution);
+        this.originalElement = Objects.requireNonNull(originalElement);
+        this.propositionFunction = Objects.requireNonNull(propositionFunction);
+        this.fetchPolicy = Objects.requireNonNull(fetchPolicy);
+    }
+
+    @Override
+    public List<RecommendedFit<Out_, Score_>> apply(InnerScoreDirector<Solution_, Score_> scoreDirector) {
+        var solutionDescriptor = scoreDirector.getSolutionDescriptor();
+        var initializationStatistics = solutionDescriptor.computeInitializationStatistics(originalSolution);
+        var uninitializedCount =
+                initializationStatistics.uninitializedEntityCount() + initializationStatistics.unassignedValueCount();
+        if (uninitializedCount > 1) {
+            throw new IllegalStateException("""
```

Comment:
Trying to understand this check. Let's say I've got the following solution with nullable variables (which is very common use case, since you might not be able to assign all the shifts everytime the solver runs).

Shift A - null
Shift B - employee1
Shift C - employee2

Now I call recommend(solution_with_shift_D, Shift D). This will contain 2 unitialized shifts (Shift A, Shift D). Does the new API handle this case, or throws an exception?

Reply:
Nullable variables count as initialized. So if the variable is nullable, the total uninitialized count will be 1.
If the variable is not nullable, the exception will be thrown, as there'd be 2 uninitialized entities.

There is another way of thinking about this - the `init` score. If the init score was anything but zero when you started, the API will fail fast.

Reply:
I edited the previous comment, as I said something that wasn't true and corrected it later.

Reply:
Sounds good, thanks for the explanation.